frontend/src/components/core/Block/utils.ts

```diff
@@ -0,0 +1,70 @@
+/**
+ * @license
+ * Copyright 2018-2021 Streamlit Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ReportRunState } from "src/lib/ReportRunState"
+import { ReportNode } from "src/lib/ReportNode"
+import { FormsData, WidgetStateManager } from "src/lib/WidgetStateManager"
+import { FileUploadClient } from "src/lib/FileUploadClient"
+import { ComponentRegistry } from "src/components/widgets/CustomComponent/"
+
+export function shouldComponentBeEnabled(
+  isHidden: boolean,
+  reportRunState: ReportRunState
+): boolean {
+  return !isHidden || reportRunState !== ReportRunState.RUNNING
+}
+
+export function isElementStale(
```

Comment:
> What's the difference between a Component and an Element, and when would you use one of these functions over the other?

I don't really know. It may not look like it, but I was trying to keep the refactor to a minimum 🤣 . This comes from the original Block class without trying to make too many changes.

Originals:
* https://github.com/streamlit/streamlit/blob/develop/frontend/src/components/core/Block/Block.tsx#L196
* https://github.com/streamlit/streamlit/blob/develop/frontend/src/components/core/Block/Block.tsx#L294

If you like I can take a stab at making other changes. I'd love to rename a lot of these types, for example, as they're super confusing...

Reply:
Ah! Yeah, sounds outside the scope of this PR. (And yeah please do rename them if you're so moved down the road!)